app/grandchallenge/profiles/templates/allauth_2fa/setup.html

```diff
@@ -0,0 +1,47 @@
+{% extends "base.html" %}
+{% load i18n %}
+
+{% block title %}Enable Two-Factor Authentication{% endblock %}
+
+{% block breadcrumbs %}
+    <ol class="breadcrumb">
+        <li class="breadcrumb-item text-light">Users</li>
+        <li class="breadcrumb-item"><a
+                href="{% url 'profile-detail' username=request.user.username %}">{{ request.user.username }}</a></li>
+        <li class="breadcrumb-item active" aria-current="page">Set-up Two Factor Authentication</li>
+    </ol>
+{% endblock %}
+
+{% block content %}
+<h1>
+  {% trans "Setup Two-Factor Authentication" %}
+</h1>
+
+<h4>
+  {% trans 'Step 1' %}:
+</h4>
+
+<p>
+  {% trans 'Scan the QR code below with a token generator of your choice (e.g., Google Authenticator, Microsoft Authenticator).' %}
+</p>
+
+<img src="{{ qr_code_url }}" />

```

Comment:
You can get the token with `get_token_from_totp_device(tot_device)` where `tot_device=user.totdevice_set.last()`

Reply:
Yes, but in order to enter it into users apps they will need to see the code in the template, rather than, or in addition to, a QR code. This is what it looks like on the GitHub Page, we would need the same (the "enter this text code" bit):

![image](https://user-images.githubusercontent.com/12661555/173315132-4edb2f2a-abfd-407b-a2a3-895eaea68ab8.png)

Reply:
Ah ok, sorry, misunderstood your comment. I'll add that!